### DISCUSSION TITLE
StrictHostKeyChecking/New SSH target in version 7

### DISCUSSION THREAD

**Discussion by mikestreety** [deployphp/deployer#3127] [Help needed]
Hello! Thanks for deployer, it's great!

I'm migrating my process to running off Gitlab CI but am running into a wall with the `ssh multiplexing initialization` stages. We have many clients with many servers, so don't want to have to embed "known_hosts" for each of the servers into the docker runner or manually specify hosts in the CI file.

I'm talking of a way to get round the following:

```
The authenticity of host 'XXX (XXX)' can't be established.
ED25519 key fingerprint is SHA256:XXX
This key is not known by any other names
Are you sure you want to continue connecting (yes/no/[fingerprint])?
```

I was wondering how I could integrate a `StrictHostKeyChecking` style option into the code?

I've tried all different versions of this to no avail (I notice a variation is mentioned in the [6.x docs](https://deployer.org/docs/6.x/hosts))

```php
->set('ssh_arguments', ['-o StrictHostKeyChecking=no'])
```

 I also wondered if I could add a custom task with a `ssh-keyscan` or `ssh XXX -oStrictHostKeyChecking=no` but I couldn't work out how to extract the user, host and port to build it.

Any help or advice would be appreciated

**Comment by antonmedv** [comment]
This should do the work:
```php
set('ssh_arguments', ['-o StrictHostKeyChecking=no']);
```

You can try to debug why it's now working with -vvv option. What ssh command is executed?

**Reply by mikestreety** [reply]
@antonmedv Thank you for the super swift reply!

This morning I realised my error, I am using [deployer-extended-typo3](https://github.com/sourcebroker/deployer-extended-typo3) and had assumed it was using version 7 when, in fact, it was using 6.

Using the documented `->addSshOption('StrictHostKeyChecking', 'no');` in 6 works as expected. Thank you!
